test/distributed/org/apache/cassandra/distributed/test/metrics/StreamingMetricsTest.java

```diff
@@ -18,105 +18,391 @@
+                .describedAs("No SSTable should have been streamed so far from node" + node + " to node" + peer)
+                .isEqualTo(0);
+
             assertThat(metrics.incomingProcessTime.getCount())
-                .isEqualTo(files)
-                .describedAs("There should be " + files + " files streamed from the peer.");
-            assertThat(metrics.incomingProcessTime.getSnapshot().getMedian())
-                .isGreaterThan(0)
-                .describedAs("The median processing time should be non-0");
-            return metrics.incomingBytes.getCount();
+                .describedAs("No SSTable should have been streamed so far from node" + node + " to node" + peer)
+                .isEqualTo(0);
         });
+    }
 
-        // Assert metrics in node 1
-        cluster.get(1).runOnInstance(() -> {
-            StreamingMetrics metrics = StreamingMetrics.get(node2Address);
-            assertThat(metrics.incomingBytes.getCount())
-                .isEqualTo(0).describedAs("There should not be sstables streamed from the peer.");
-            assertThat(metrics.outgoingBytes.getCount())
-                .isEqualTo(transmittedBytes)
-                .describedAs("The outgoingBytes count in node1 should be equals to incomingBytes count in node2");
-            assertThat(metrics.incomingProcessTime.getCount())
-                .isEqualTo(0)
-                .describedAs("There should be no files streamed from the peer.");
+    private long checkDataSent(Cluster cluster, int node, int peer)
+    {
+        InetAddressAndPort address = getNodeAddress(cluster, peer);
+        return cluster.get(node).callOnInstance(() -> {
+
+            StreamingMetrics metrics = StreamingMetrics.get(address);
+
+            long outgoingBytes = metrics.outgoingBytes.getCount();
+            assertThat(outgoingBytes)
+                .describedAs("There should be data streamed from node" + node + " to node" + peer)
+                .isGreaterThan(0);
+
+            return outgoingBytes;
         });
     }
+
+    private void checkDataReceived(Cluster cluster, int node, int peer, long receivedBytes, int files)
+    {
+        InetAddressAndPort address = getNodeAddress(cluster, peer);
+        cluster.get(node).runOnInstance(() -> {
+
+            StreamingMetrics metrics = StreamingMetrics.get(address);
+
+            long actual = metrics.incomingBytes.getCount();
+            assertThat(actual)
+                .describedAs("The amount of data received by node" + node + " from node" + peer + " is not the expected one. [expected: " + receivedBytes + ", actual: " + actual + "]")
+                .isEqualTo(receivedBytes);
+
+            actual = metrics.incomingProcessTime.getCount();
+            assertThat(actual)
+                .describedAs("The amount of files received by node" + node + " from node" + peer + " is not the expected one. [expected: " + files + ", actual: " + actual + "]")
+                .isEqualTo(files);
```

Comment:
nit: I would probably comment how we ended up comparing  incomingProcessTime.getCount() to files, for clarity